Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTip.cs

```diff
@@ -0,0 +1,540 @@
+        /// <summary>
+        /// getter/setter for ameObject that the line and text are attached to
+        /// </summary>
+        public GameObject Anchor
+        {
+            get
+            {
+                return anchor;
+            }
+            set
+            {
+                anchor = value;
+            }
+        }
+
+        [Tooltip("Pivot point that text will rotate around as well as the point where the Line will be rendered to.")]
+        [SerializeField]
+        private GameObject pivot;
+        /// <summary>
+        /// Pivot point that text will rotate around as well as the point where the Line will be rendered to. 
+        /// </summary>
+        public GameObject Pivot
+        {
+            get
+            {
+                return pivot;
+            }
+        }
+
+        [Tooltip("GameObject text that is displayed on the tooltip.")]
+        [SerializeField]
+        private GameObject label;
+
+        [Tooltip("Parent of the Text and Background")]
+        [SerializeField]
+        private GameObject contentParent;
+
+        [Tooltip("Text for the ToolTip to say")]
+        [SerializeField]
+        [TextArea]
+        private string toolTipText;
+        /// <summary>
+        /// Text for the ToolTip to display
+        /// </summary>
+        public string ToolTipText
+        {
+            set
+            {
+                if (value != toolTipText)
+                {
+                    toolTipText = value;
+                    RefreshLocalContent();
+                    if (ContentChange != null)
+                        ContentChange.Invoke();
+                }
+            }
+            get
+            {
+                return toolTipText;
+            }
+        }
+
+        [Tooltip("The padding around the content (height / width)")]
+        [SerializeField]
+        private Vector2 backgroundPadding;
+
+        [Tooltip("The offset of the background (x / y / z)")]
+        [SerializeField]
+        private Vector3 backgroundOffset;
+        /// <summary>
+        /// The offset of the background (x / y / z)
+        /// </summary>
+        public Vector3 LocalContentOffset
+        {
+            get
+            {
+                return backgroundOffset;
+            }
+        }
+
+        [Tooltip("The scale of all the content (label, backgrounds, etc.)")]
+        [SerializeField]
+        [Range(0.01f, 3f)]
+        private float contentScale = 1f;
+        /// <summary>
+        /// The scale of all the content (label, backgrounds, etc.)
+        /// </summary>
+        public float ContentScale
+        {
+            get
+            {
+                return contentScale;
+            }
+            set
+            {
+                contentScale = value;
+                RefreshLocalContent();
+            }
+        }
+
+        [Tooltip("The font size of the tooltip.)")]
+        [SerializeField]
+        [Range(10, 60)]
+        private int fontSize = 30;
```

Comment:
Allow apps to change this value at runtime? Or would that cause downstream issues?